src/java.security.jgss/share/native/libj2gss/NativeUtil.c

```diff
@@ -724,66 +786,47 @@ jobject getJavaOID(JNIEnv *env, gss_OID cOid) {
   if (jbytes == NULL) {
     return NULL;
   }
-  (*env)->SetByteArrayRegion(env, jbytes, 0, 2, (jbyte *) oidHdr);
-  if ((*env)->ExceptionCheck(env)) {
-    return NULL;
+  if (!(*env)->ExceptionCheck(env)) {
```

Comment:
Consider using `GetPrimitiveArrayCritical()` instead. I think it would result in simpler, faster code.

Reply:
Hmm, but can you write to an array using GetPrimitiveArrayCritical()?